src/templates/next/components/Navbar/Navbar.tsx

```diff
@@ -1,178 +1,285 @@
-import { Fragment } from "react"
-import { Disclosure, Menu, Transition } from "@headlessui/react"
-
+import { useState } from "react"
 import {
-  Bars3Icon,
-  XMarkIcon,
-  MagnifyingGlassIcon,
-} from "@heroicons/react/24/outline"
-
-export type NavbarLink = {
-  type: "single" | "dropdown"
-  name: string
-  eventKey?: string
-  url?: string
-  links?: NavbarLink[]
-}
+  BiChevronDown,
+  BiChevronRight,
+  BiChevronUp,
+  BiLeftArrowAlt,
+  BiMenu,
+  BiRightArrowAlt,
+  BiSearch,
+  BiX,
+} from "react-icons/bi"
+import type { NavbarProps } from "~/common"
 
-function classNames(...classes: any) {
-  return classes.filter(Boolean).join(" ")
-}
+const Navbar = ({
+  logoUrl,
+  logoAlt,
+  search,
+  items,
+  LinkComponent = "a",
+}: NavbarProps) => {
+  const [navItemOpen, setNavItemOpen] = useState(-1)
```

Comment:
could we rename to `currentlyOpenNavItem` or `openNavItemIdx`? reads abit oddly now